sympy/physics/control/lti.py

```diff
@@ -1422,3 +1445,551 @@ def _eval_rewrite_as_TransferFunction(self, num, den, **kwargs):
+        >>> tfm_2
+        TransferFunctionMatrix(((TransferFunction(10, 1, s), TransferFunction(1, s*(s - 1), s)), (TransferFunction(p, s - 1, s), TransferFunction(a, a*s + p, s))))
+        >>> pprint(tfm_2, use_unicode=False)
+        [ 10        1    ]
+        [ --    ---------]
+        [ 1     s*(s - 1)]
+        [                ]
+        [  p        a    ]
+        [-----   ------- ]
+        [s - 1   a*s + p ]
+
+        """
+        return _to_TFM(matrix, var)
+
+    @property
+    def var(self):
+        """
+        Returns the complex variable used by all the transfer functions or
+        ``Series``/``Parallel`` objects in a transfer function matrix.
+
+        Examples
+        ========
+
+        >>> from sympy.abc import p, s
+        >>> from sympy.physics.control.lti import TransferFunction, TransferFunctionMatrix, Series, Parallel
+        >>> G1 = TransferFunction(p**2 + 2*p + 4, p - 6, p)
+        >>> G2 = TransferFunction(p, 4 - p, p)
+        >>> G3 = TransferFunction(0, p**4 - 1, p)
+        >>> G4 = TransferFunction(s + 1, s**2 + s + 1, s)
+        >>> S1 = Series(G1, G2)
+        >>> S2 = Series(-G3, Parallel(G2, -G1))
+        >>> tfm1 = TransferFunctionMatrix([[G1], [G2], [G3]])
+        >>> tfm1.var
+        p
+        >>> tfm2 = TransferFunctionMatrix([[-S1, -S2], [S1, S2]])
+        >>> tfm2.var
+        p
+        >>> tfm3 = TransferFunctionMatrix([[G4]])
+        >>> tfm3.var
+        s
+
+        """
+        return self.args[0][0][0].var
+
+    @property
+    def num_inputs(self):
+        """
+        Returns the number of inputs of the system.
+
+        Examples
+        ========
+
+        >>> from sympy.abc import s, p
+        >>> from sympy.physics.control.lti import TransferFunction, TransferFunctionMatrix
+        >>> G1 = TransferFunction(s + 3, s**2 - 3, s)
+        >>> G2 = TransferFunction(4, s**2, s)
+        >>> G3 = TransferFunction(p**2 + s**2, p - 3, s)
+        >>> tfm1 = TransferFunctionMatrix([[G1], [G2]])
+        >>> tfm1.num_inputs
+        1
+        >>> tfm2 = TransferFunctionMatrix([[G2, -G1, G3], [-G2, -G1, -G3]])
+        >>> tfm2.num_inputs
+        3
+
+        """
+        return self._expr_mat.shape[1]
+
+    @property
+    def num_outputs(self):
+        """
+        Returns the number of outputs of the system.
+
+        Examples
+        ========
+
+        >>> from sympy.abc import s, p, a, b
+        >>> from sympy.physics.control.lti import TransferFunction, Series, TransferFunctionMatrix
+        >>> tf1 = TransferFunction(a*p**2 + b*s, s - p, s)
+        >>> tf2 = TransferFunction(s**3 - 2, s**4 + 5*s + 6, s)
+        >>> tf3 = TransferFunction(a*s - 4, s**4 + 1, s)
+        >>> S1, S2 = Series(tf1, -tf2), Series(tf2, tf3, -tf1)
+        >>> TFM1 = TransferFunctionMatrix([[tf1], [tf2], [tf3]])
+        >>> TFM1.num_outputs
+        3
+        >>> TFM2 = TransferFunctionMatrix([[S1, S2], [-S2, -S1]])
+        >>> TFM2.num_outputs
+        2
```

Comment:
I would aim for the simplest demonstration possible here:
```
>>> TFM = TransferFunction.from_Matrix([[s], [1/s]], s)
>>> print(TFM)
>>> TFM.num_outputs
2
```
Multiple examples are hardly needed for something this simple. What would be more useful is an explicit statement that this is just the number of rows in the matrix.

This could have a see also for `num_inputs`